autoload/go/debug.vim

```diff
@@ -381,6 +381,7 @@ function! go#debug#Start(...) abort
   try
     echohl SpecialKey | echomsg 'Starting GoDebug...' | echohl None
     let s:state['message'] = []
+    exe 'lcd' fnamemodify(bufname(''), ':p:h')
```

Comment:
Won't this cause the current working directory to change? `:pwd` gives me `/home/martin/go/src/a` when starting `vim dbg/debug.go`, but it's `/home/martin/go/src/a/debug` after running `:GoDebugStart`.

Also, I get an error using `:GoDebugStart` with the latest changes?

```
can't load package: package ['--']: cannot find package "['--']" in any of:
        /usr/lib/go/src/['--'] (from $GOROOT)
        /home/martin/go/src/['--'] (from $GOPATH)
        /home/martin/work/src/['--']
exit status 1
```

Maybe it's worth adding some tests for this? I never really looked in to testing Vimscript stuff so I don't know how hard that is.

Reply:
fixed for `["--"]`.

`lcd` change current directory for current window.